### PR TITLE
feat: Add custom exception messages

### PR DESCRIPTION
**Pull request by dinispeixoto**
# PR Details

- Raise ValueError instead of generic Exception
- Create Custom Exceptions for each validation
- Update folder structure

## Motivation and Context

- To add Custom exception messages
- `ValueError` instead of `Exception`. `ValueError(Exception)` can be trapped alone i.e. `except(ValueError)` is more useful than `except(Exception)`

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by claudiocn — CHANGES_REQUESTED** [review]
Please change the documentation to use the "errors" instead of the old "exceptions"

File: checkarg/exceptions/error_messages.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    @staticmethod
+    def is_lower_or_equals_message(argument_name, value, condition_value):
+        return f"Argument {argument_name} with value {value} should be lower than or equal to {condition_value}"
+
+
+class TextErrorMessages:
+    @staticmethod
+    def is_not_whitespace_message(argument_name):
+        return f"Argument {argument_name} shouldn't be whitespace"
```
Comment (claudiocn) [inline checkarg/exceptions/error_messages.py:34]: "Shouldn't be only whitespaces"

Comment (claudiocn) [inline checkarg/exceptions/error_messages.py:34]: "Shouldn't be only whitespaces"

File: checkarg/exceptions/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from .error_messages import DefaultErrorMessages
+
+# ValueError instead of Exception
+# ValueError(Exception) but can be trapped alone
+# i.e. except(ValueError) is more useful than except(Exception)
```
Comment (claudiocn) [inline checkarg/exceptions/exceptions.py:5]: remove this comments

Comment (claudiocn) [reply checkarg/exceptions/exceptions.py:5]: And add this idea to PR description.

Comment (claudiocn) [reply checkarg/exceptions/exceptions.py:5]: And Use PEP8 as motivation.

File: setup.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 from typing import List
+
```
Comment (claudiocn) [inline setup.py:2]: Confirm if this was done by the formatter, if so keep this change.

**Review by claudiocn — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (dinispeixoto) [reply setup.py:2]: yap, it was

**Review by claudiocn — CHANGES_REQUESTED** [review]
Seems really cool change. I would only request a version change because it adds functionality and it doesn't have breaking changes, so a "minor" should be. :)

**Review by claudiocn — APPROVED** [review]

**Pull request merged**